pkg/util/controllerfinder/pods_finder.go

```diff
@@ -102,8 +99,34 @@ func (r *ControllerFinder) GetPodsForRef(apiVersion, kind, ns, name string, acti
 			}
 			matchedPods = append(matchedPods, pod)
 		}
+		return matchedPods, nil
+	}
+
+	var err error
+	var matchedPods []*corev1.Pod
+	for _, uid := range workloadUIDs {
+		listOption := client.ListOptions{
+			Namespace:     ns,
+			FieldSelector: fields.SelectorFromSet(fields.Set{fieldindex.IndexNameForOwnerRefUID: string(uid)}),
+		}
+		pods, err := listPods(&listOption)
+		if err != nil {
+			return nil, -1, err
+		}
+		matchedPods = append(matchedPods, pods...)
 	}
 
+	if labelSelector != nil && len(matchedPods) == 0 { // try again using label selector for scale sub-resource
```

Comment:
why retry using the pod label selector ? plz comment about the cases

Reply:
done